sbot/utils.py

```diff
@@ -0,0 +1,201 @@
+"""General utility functions and classes for the sbot package."""
+from __future__ import annotations
+
+import logging
+import socket
+from abc import ABC, abstractmethod
+from typing import Any, Mapping, NamedTuple, TypeVar
+
+from serial.tools.list_ports_common import ListPortInfo
+
+T = TypeVar('T')
+logger = logging.getLogger(__name__)
+
+
+class BoardIdentity(NamedTuple):
+    """
+    A container for the identity of a board.
+
+    All the board firmwares should return this information in response to
+    the *IDN? query.
+
+    :param manufacturer: The manufacturer of the board
+    :param board_type: The short name of the board, i.e. PBv4B
+    :param asset_tag: The asset tag of the board,
+        this should match what is printed on the board
+    :param sw_version: The firmware version of the board
+    """
+    manufacturer: str = ""
+    board_type: str = ""
+    asset_tag: str = ""
+    sw_version: str = ""
```

Comment:
Suggestion: Might be nice to remove the defaults and force each argument to be defined. We're already doing that in the majority of cases, so the defaults are redundant.

Reply:
It makes some of the bits around USB generated identities and fucntion default values very verbose. In both these cases we normally don't have all the fields.